test/selenium/README.md

```diff
@@ -22,20 +22,80 @@ Run the [run-tests script](https://github.com/cockpit-project/cockpit/blob/maste
 
 Although this is the default way to run selenium tests the run-tests script is configurable and can be changed to run tests against different machines. This can be useful for developing or debugging tests. Check bellow the HACKING section for more details.
 
+### Debugging tests:
+Selenium testsuite via ``run-tests``allows the possibility of how to debug tests.
+  - Let machines running via ``--sit`` option what let all machines running after tests finish
+  - Run just selected (one or more) tests via listing them on the command line, tests are relative to ``test/selenium/`` directory
+  - Use own selenium grid via option ``--hub``
+
+#### Use selenium grid in container
+Start own selenium grid in container ([Original description](https://github.com/SeleniumHQ/docker-selenium#debugging)).
+
+  - Create selenium grid with chrome browser with VNC enabled (you can replace it by firefox and change commands according to that):
+    ```
+    podman run --rm -p 4444:4444 -p 5555:5900 -v /dev/shm:/dev/shm selenium/standalone-chrome-debug
+    ```
+  - Start VNC session to that machine (password is ``secret``):
+    ```
+    vncviewer 127.0.0.1:5555
+    ```
+  - Enable port forwarding of port ``127.0.0.2:9091`` to publicly accessible address, because selenium hub in podman container will have trouble to contact host port, because it is forwarded right to ``127.0.0.2``
+    ```
+    ssh -L $(hostname -I | cut -d ' ' -f 1):9991:127.0.0.2:9091 $(whoami)@localhost
+    ```
+  - Run the test
+    ```
+    test/selenium/run-tests --hub $(hostname -I | cut -d ' ' -f 1):$(hostname -I | cut -d ' ' -f 1):9991 -v -b chrome selenium-base.py --sit
+    ```
+
+#### Selenium grid on your machine with your local browsers
+Steps are described in [selenium page](https://www.browserstack.com/guide/selenium-grid-tutorial)
+  - Download selenium standalone:
+    ```
+    curl -f -L https://bit.ly/2TlkRyu > selenium.jar
+    ```
+  - Run hub and attach browsers here:
+    ```
+    java -jar selenium.jar -role hub
+    ```
+    - Then attach your local browsers there. You have browsers eg. firefox or chrome installed, and also ensure you have proper [selenium drivers for selected browser](https://github.com/cockpit-project/cockpit/tree/master/test/selenium#which-browser-to-use)
+      ```
+      java -jar selenium.jar -role node -hub https://localhost:4444/grid/register
+      ```
+  - Run the test
+    ```
+    test/selenium/run-tests --hub $(hostname -I | cut -d ' ' -f 1) -v -b chrome selenium-base.py --sit
+    ```
+#### Rerun the the test/s again
+It is enabled when you'll use ``--sit`` option.
+Command-line options depend on which selenium grid you used.
+It is described on the stderr output of the ``run-tests`` script.
+
+It is possible also to run the tests on your machine, without avocado machine.
```

Comment:
What's an avocado machine?
You mean the test VMs, with images gotten from bots/images/?